core/engine/src/vm/opcode/get/property.rs

```diff
@@ -164,6 +249,12 @@ impl GetPropertyByValuePush {
                 context.vm.set_register(dst.into(), element);
                 return Ok(());
             }
+        } else if let Some(value) = js_string_get(&base, &key_value) {
+            context
+                .vm
+                .set_register(key.into(), StaticJsStrings::LENGTH.into());
```

Comment:
Is this correct? `js_string_get` could also get an index property that is not "length", right?

Reply:
Nice catch! That's a mistake, fixed in https://github.com/boa-dev/boa/pull/4452/commits/72b677aa456ed7e363e14ff0e5943b7e156d3ab9